core/src/main/scala/org/apache/spark/sql/delta/commands/DeleteCommand.scala

```diff
@@ -81,6 +81,29 @@ case class DeleteCommand(
 
   override lazy val metrics = createMetrics
 
+  sealed trait RowDeletion
+
+  case class RowInFilesDeletion(rowCount: Long) extends RowDeletion
+
+  case object FilePruningAtPartitionBoundary extends RowDeletion
+
+  private def deletionResultFromMetrics: RowDeletion = {
+    val deletedRows = this.metrics.get("numDeletedRows").map(_.value).getOrElse(0L)
+    if(deletedRows > 0) {
+      RowInFilesDeletion(deletedRows)
+    } else {
+      val deletedFiles = this.metrics.get("numRemovedFiles").map(_.value).getOrElse(0L)
```

Comment:
We can also get the metric value using just `metrics("numRemovedFiles").value` and `metrics("numDeletedRows").value`. This is how we do it throughout the code, and we see above that both keys are defined in `createMetrics`. And as @scottsand-db mentioned the default value is 0 anyways.

Reply:
> Don't you think that accessing a map directly relying on the fact that same values are always present is relying on some implicit knowledge about the content of the map that is not encoded within the type system and which can also change leading to failures?

I think we would want to know if the content of our metrics map changed to not contain these values that we explicitly define.

Reply:
@allisonport-db are there  unit tests that verify that certain metrics are set by 0 to default?

Reply:
I'm not sure if there's a test for this as `SQLMetric` is also a spark class. We can see [in the class header](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala#L40) the default value is 0.

We can test this to some degree however by, as I've mentioned, adding a test for when `numDeletedRows=0`.